operator/pkg/metrics/monitoring.go

```diff
@@ -0,0 +1,240 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"istio.io/pkg/monitoring"
+)
+
+// OperatorVersionLabel describes version of running binary
+var OperatorVersionLabel = monitoring.MustCreateLabel("v")
+
+// MergeErrorLabel describes the type of merge error
+var MergeErrorLabel = monitoring.MustCreateLabel("error_type")
+
+// PathLabel describes translated path
+var PathLabel = monitoring.MustCreateLabel("path")
+
+// RenderErrorLabel describes the type of the error while rendering
+var RenderErrorLabel = monitoring.MustCreateLabel("render_error")
+
+var (
+	// CRFetchErrorReasonLabel describes the reason/HTTP code
+	// for failing to fetch CR
+	CRFetchErrorReasonLabel = monitoring.MustCreateLabel("reason")
+
+	// CRNamespacedNameLabel describes the name with namespace
+	CRNamespacedNameLabel = monitoring.MustCreateLabel("name")
+)
+
+var (
+	// ComponentNameLabel represents istio component name - like
+	// core, pilot, istio-cni etc.
+	ComponentNameLabel = monitoring.MustCreateLabel("component")
+
+	// ResourceKindLabel indicates the kind of resource owned
+	// or created or updated or deleted or pruned by operator
+	ResourceKindLabel = monitoring.MustCreateLabel("kind")
+
+	// CRRevisionLabel indicates the revision in the CR
+	CRRevisionLabel = monitoring.MustCreateLabel("revision")
+)
+
+// MergeErrorType describes the class of errors that could
+// occur while merging profile, user supplied YAML, values
+// overridden by --set and so on
+type MergeErrorType string
+
+const (
+	// CannotFindProfileError occurs when profile cannot be found
+	CannotFindProfileError MergeErrorType = "fetch_profile"
+
+	// CannotMergeProfileWithHubTagError occurs when merging profile
+	// and Hub-Tag overlay fails
+	CannotMergeProfileWithHubTagError MergeErrorType = "overlay_hub_tag_on_profile"
+
+	// CannotMarshalUserIOPError occurs when user supplied CR cannot
+	// be marshaled to YAML string
+	CannotMarshalUserIOPError MergeErrorType = "user_iop_marshal"
+
+	// CannotMergeFileOverlayWithProfileError occurs when file overlay
+	// supplied by user cannot be overlaid on profile+hub/tag overlay
+	CannotMergeFileOverlayWithProfileError MergeErrorType = "overlay_file_on_profile"
+
+	// CannotGetSpecSubtreeError occurs when spec section in merged CR
+	// cannot be accessed for some reason (either missing or multiple)
+	CannotGetSpecSubtreeError MergeErrorType = "spec_subtree"
```

Comment:
see comments elsewhere - these are probably too detailed for a metric.

Reply:
Resolved. Renamed as per your suggestion